src/fold/base/classes.py

```diff
@@ -424,4 +424,4 @@ def from_events_sample_weights(
         if result.empty:
             return Artifact.empty(index)  # type: ignore
         else:
-            return result  # type: ignore
+            return result.reindex(index)  # type: ignore
```

Comment:
this will epxand artifacts to have the same shape as X, priori to passing it into the pipeline